src/components/DpAutocomplete/DpAutocomplete.vue

```diff
@@ -1,114 +1,396 @@
+      </ol>
+    </div>
+  </div>
 </template>
 
-<script>
-import { dpApi } from '~/lib'
-import { prefixClassMixin } from '~/mixins'
+<script setup lang="ts">
+import { computed, nextTick, onMounted, ref, watch } from 'vue'
 import { de } from '~/components/shared/translations'
-import VueOmnibox from 'vue-omnibox'
+import { dpApi } from '~/lib/DpApi'
 
-export default {
-  name: 'DpAutocomplete',
+/**
+ * This component provides an autocomplete input field with dropdown suggestions.
+ * It fetches suggestions based on user input via an API call.
+ */
 
-  components: {
-    VueOmnibox
+// Props
+const props = defineProps({
+  /**
+   * Height of the autocomplete box
+   */
+  height: {
+    type: String,
+    required: false,
+    default: '28px'
   },
 
-  mixins: [prefixClassMixin],
-
-  props: {
-    height: {
-      type: String,
-      required: false,
-      default: '28px'
-    },
-
-    label: {
-      type: String,
-      required: false,
-      default: 'label'
-    },
-
-    options: {
-      type: Array,
-      required: false,
-      default: () => ([])
-    },
-
-    placeholder: {
-      type: String,
-      required: false,
-      default: de.placeholderAutoSuggest
-    },
-
-    routeGenerator: {
-      type: Function,
-      required: true
-    },
-
-    value: {
-      type: String,
-      required: false,
-      default: ''
-    }
+  /**
+   * The property name from the options objects to display
+   */
+  label: {
+    type: String,
+    required: false,
+    default: 'label'
   },
 
-  data () {
-    return {
-      currentQuery: this.value,
-      isLoading: false
-    }
+  /**
+   * Array of options to display in the dropdown
+   */
+  options: {
+    type: Array,
+    required: false,
+    default: () => ([])
   },
 
-  watch: {
-    value () {
-      if (this.currentQuery !== this.value) {
-        this.currentQuery = this.value
-      }
-    }
+  /**
+   * Placeholder text to display when the input is empty
+   */
+  placeholder: {
+    type: String,
+    required: false,
+    default: de.placeholderAutoSuggest
   },
 
-  methods: {
-    emit (e, type) {
-      this.$emit(type, e)
-    },
-
-    emitAndFetch (value) {
-      this.emit(value, 'input')
-      this.currentQuerry = value
-      if (value.length >= 3) this.fetchOptions(value)
-    },
-
-    async fetchOptions (searchString) {
-      this.isLoading = true
-      try {
-        const route = this.routeGenerator(searchString)
-        const response = await dpApi.get(route)
-        // Only emit results that match the current search -> prevents race conditions
-        if (this.currentQuerry === searchString) this.$emit('search-changed', response)
-        this.isLoading = false
-      } catch (e) {
-        console.log(e)
+  /**
+   * Function that generates the API route for fetching suggestions
+   */
+  routeGenerator: {
+    type: Function,
+    required: true
+  },
+
+  /**
+   * Current value of the input (v-model)
+   */
+  modelValue: {
```

Comment:
I think the core should also be updated to use the model-value prop instead of v-model, because `currentAutocompleteSearch` remains empty. So that clicking the save button produces no results. Is there a PR for this?

Reply:
can't reprocude, what You mean. For me it works when I test it in public_index or public_detail. Even without modelValue.
But You are right. I should adjust it in core too..